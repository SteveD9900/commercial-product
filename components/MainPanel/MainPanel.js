import React, { useState } from "react";
import Card from "../Card/Card";

import styles from "./MainPanel.module.scss";

export default function MainPanel(props) {

  return (
    <div className={styles.MainPanel}>
        <div className={styles.productContainer}>
        {props.products.map((product) => (
          <Card key={product.index} details={product}/>
        ))}
        {props.products.length === 0 ? (
            <span>No Result</span>
        ) : null}
        </div>
    </div>
  );
}
