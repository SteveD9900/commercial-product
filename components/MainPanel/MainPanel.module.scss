.MainPanel {
    width: 100%;
    height: auto;
}

.productContainer {
    width: 80%;
    height: auto;
    margin: auto;
    border: 1px solid black;
    display: grid;
    grid-template-columns: repeat(auto-fit, minmax(305px, 1fr));
    grid-gap: 1px;
}

