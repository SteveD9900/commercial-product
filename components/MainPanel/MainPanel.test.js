import MainPanel from './MainPanel';
import { shallow } from 'enzyme'

let wrapper;
let filteredProducts = [
    {
        "index": 0,
        "isSale": false,
        "price": "$49.99",
        "productImage": "Product_1.jpeg",
        "productName": "Pure Blonde Crate",
        "type": "Beer"
    },
    {
        "index": 1,
        "isSale": true,
        "price": "$14.99",
        "productImage": "Product_2.jpeg",
        "productName": "Victoria Bitter 4x6x375ml",
        "type": "Beer"
    }
];

beforeEach(() => {
    wrapper = shallow(<MainPanel products={filteredProducts}/>);
});

describe('<MainPanel /> rendering', () => {
    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});