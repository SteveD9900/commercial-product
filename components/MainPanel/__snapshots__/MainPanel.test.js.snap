// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainPanel /> rendering renders correctly 1`] = `ShallowWrapper {}`;
